packages/esm-patient-list-app/src/api/hooks.ts

```diff
@@ -144,3 +161,97 @@ export function useCohortTypes() {
   const swrResult = useSWR<FetchResponse<CohortResponse<CohortType>>, Error>(`${cohortUrl}/cohorttype`, openmrsFetch);
   return { ...swrResult, data: swrResult?.data?.data?.results };
 }
+
+export function usePatientListIdsForPatient(patientUuid: string) {
+  const { data, isLoading, isValidating } = useSWR<FetchResponse<CohortResponse<OpenmrsCohortRef>>>(
+    `${cohortUrl}/cohortmember?patient=${patientUuid}&v=default`,
+    openmrsFetch,
+  );
+
+  return {
+    listsIdsOfThisPatient: data?.data?.results.map((ref) => ref.cohort.uuid),
+    isLoadingPatientListIdsForPatient: isLoading,
+    isvalidatingPatientListIdsForPatient: isValidating,
+  };
+}
+
+export function useAddablePatientLists(patientUuid: string, name: string = '') {
+  const { t } = useTranslation();
+  const {
+    patientLists: realPatientLists,
+    isLoading: isLoadingRealLists,
+    isValidating: isValidatingAllLists,
+  } = useAllPatientLists({ name });
+
+  const { listsIdsOfThisPatient, isLoadingPatientListIdsForPatient, isvalidatingPatientListIdsForPatient } =
+    usePatientListIdsForPatient(patientUuid);
+
+  const {
+    data: fakePatientLists,
+    isLoading: isLoadingFakeLists,
+    isValidating: isValidatingFakeLists,
+    error,
+  } = useSWR('fakePatientLists', async () => await findFakePatientListsWithoutPatient(patientUuid, t));
+
+  const addableRealPatientList = (() => {
+    if (!fakePatientLists || !realPatientLists || !listsIdsOfThisPatient) {
+      return [];
+    }
+    return realPatientLists.map((list) => ({
+      id: list.id,
+      displayName: list.display,
+      checked: listsIdsOfThisPatient.includes(list.id),
+      async addPatient() {
+        await addPatientToList({
+          cohort: list.id,
+          patient: patientUuid,
+          startDate: toOmrsIsoString(new Date()),
+        });
+      },
+    }));
+  })();
+
+  const addableLists =
+    fakePatientLists && realPatientLists && listsIdsOfThisPatient
+      ? [
+          ...(name
+            ? fakePatientLists.filter((list) => list.displayName.toLowerCase().includes(name))
+            : fakePatientLists),
+          ...addableRealPatientList,
+        ]
+      : [];
+
+  return {
+    addableLists: addableLists,
+    isLoadingLists: isLoadingRealLists || isLoadingFakeLists || isLoadingPatientListIdsForPatient,
+    isValidating: isValidatingAllLists || isValidatingFakeLists || isvalidatingPatientListIdsForPatient,
+    error,
```

Comment:
The error can come on any of the 3 hooks.
`error: all three error objects`
Please also do the error handling as well.